open-notes/maturity-model.md

```diff
@@ -0,0 +1,394 @@
+---
+id: '202107270936'
+tags: []
+related: []
+from:
+---
+
+# Maturity Model
+
+This is something that each org or team could come up by themselves depending on the their objectives. My feeling is that in an organisation having these standard across teams of the same type ie. dev, infra, etc. could be beneficial for standardizing in a way that still fosters autonomy.
+
+This goes toward something I believe quite strongly. Standardizing when you have immature standards is a good way to get stuck in mediocrity, at best. Unfortunately, you cannot know what you do not know, so my hope is that publishing and crowd-sourcing these maturity models across organisations will lead to better outcomes.
+
+A maturity model is made up of levels. Each of these levels have various skills in them. I find skill a good word for these as usually anything that brings benefit to a team requires practice. So the skill is not just about doing something, but practicing it until you are skillful at it.
+Take unit testing as an example. You can have some unit tests in your project and say you have that Skill. Having unit tests in your project is different from having maintainable unit tests that test useful aspects of your codebase. If they are written mostly after the fact and submitted as completely different PRs, they are not really helping to catch bugs. This is a sign of chasing a test coverage metric rather than using unit testing as a skill to deliver higher quality software.
+
+And here in lies the danger in having a checklist like this maturity model. It can be viewed as a todo list to check-off, maybe even a source of pride over other teams. This is not the intent! It is meant as a guide to becoming a more skillful team at delivering quality software that is a joy to work with and adds value to customers.
```

Comment:
I think there's something about this way of organising. You have chosen Level as the primary dimension instead of category/goal. There are certain advantages to that but I personally see also some disadvantages:

1. It inadvertently puts the primary focus on obtaining levels instead on obtaining goals. This is clearly not your intention, but I anticipate this to be an effect of placing it first.
2. It suggests that there is a sequence of sorts to go through. There's some truth to that on the team level, but that structure becomes more vague the larger the scope (organisation). If you want each team to define their own maturity model, that could work, if you want to standardise across the organisation, this could hurt.
3. The levels themselves aren't so meaningful to me. I imagine finding particular groupings of skills within levels for an organisation or a team is a cumbersome task with too much room for philosophical debate. 

In conclusion, I think you need to carefully balance standardising the maturity model across the organisation (with meaningful cross-team metrics) against the specific maturity demands the context places upon a team (with implied ordering)

An alternative you could consider:
1. Make the goals (security, healthy process, knowledge distribution, etc.) the primary axis. This will emphasise focusing on goals and creating an appropriate profile. To make it more visual: I think of maturity more as a [radar](https://www.researchgate.net/figure/Radar-chart-with-Safety-Dynamics-components-for-various-factors_fig2_327562127) as opposed to a ladder. You can still enforce a ladder by taking the min/median/max/sum over the level within all of the goals.
2. Use a point based system to measure maturity / skill application. Each team defines the number of points obtained for a practicing a particular skill, as relevant to their own context. This is also an indirection like levels but with does not demand introducing ordering in the maturity evaluation. I think the level ordering isn't too useful. Ordering is either demanded already by 'reality' (i.e. to have business monitoring, there are certain prerequisites which you need to meet). Or the ordering is just an expression of (more or less) value, but then obtaining the lesser value is still of value, even if you didn't acquire the greater one first (i.e. a refinement is still valuable even if you don't do retro's yet, and vice versa)

An effect of a system following the above is that you can easily visualise a team's maturity against their own valuation, but also against an organisational-average valuation (or just compare valuations directly). I think these form convenient/good discussion starters about why valuations differ. It balances well the prescriptive nature of standards with the necessities mandated by each teams' context.

Sorry for the long rant :) Hope it helps in some form!